core/src/local/persistent_state.rs

```diff
@@ -410,28 +406,29 @@ impl PersistentState {
     // with exactly those values. I think the regular state implementation supports inserting
     // something like an Int and retrieving with a BigInt.
     fn insert(&mut self, batch: &mut WriteBatch, r: &[DataType]) {
-        for index in self.indices.iter_mut() {
+        for index in self.indices[1..].iter_mut() {
             index.seq += 1;
         }
 
-        let (secondary_indices, serialized_pk) = if let Some(ref pk_cols) = self.primary_key {
-            let pk = KeyType::from(pk_cols.iter().map(|i| &r[*i]));
-            (&self.indices[..], Self::serialize_prefix(0, &pk))
-        } else {
-            // For bases without primary keys we store the actual row values keyed by the index
-            // that was added first. This means that we can't consider the keys unique though, so
-            // we'll append a sequence number.
-            let pk_index = &self.indices[0];
-            let pk = KeyType::from(pk_index.columns.iter().map(|i| &r[*i]));
-            (&self.indices[1..], self.serialize_key(0, &pk, pk_index.seq))
+        let serialized_pk = {
+            let pk = KeyType::from(self.indices[0].columns.iter().map(|i| &r[*i]));
+            if self.has_unique_index {
+                Self::serialize_prefix(0, &pk)
+            } else {
+                // For bases without primary keys we store the actual row values keyed by the index
+                // that was added first. This means that we can't consider the keys unique though, so
+                // we'll append a sequence number.
+                self.indices[0].seq += 1;
```

Comment:
Given the loop further up that adds 1 to all `seq`s, why is it necessary to add again here?

Reply:
> Don't you need to increment seq for the first index if it's not a primary key too?

That happens here. Might be more explicit to have a conditional in the for loop though, to avoid spreading out the logic.

Reply:
Ohhh, I missed the `[0]` part. My bad.